File: app/src/main/java/org/autojs/autojs/ui/common/ScriptLoopDialog.java
package org.autojs.autojs.ui.common;

import android.content.Context;
import androidx.annotation.NonNull;

import android.view.LayoutInflater;
import android.view.View;
import android.view.Window;
import android.widget.EditText;

import com.afollestad.materialdialogs.DialogAction;
import com.afollestad.materialdialogs.MaterialDialog;
import com.stardust.app.DialogUtils;
import com.stardust.app.GlobalAppContext;

import org.autojs.autojs.R;
import org.autojs.autojs.databinding.DialogScriptLoopBinding;
import org.autojs.autojs.model.script.ScriptFile;
import org.autojs.autojs.model.script.Scripts;

public class ScriptLoopDialog {

    private ScriptFile mScriptFile;
    private MaterialDialog mDialog;
    private DialogScriptLoopBinding binding;

    public ScriptLoopDialog(Context context, ScriptFile file) {
        mScriptFile = file;
        binding = DialogScriptLoopBinding.inflate(LayoutInflater.from(context));

        mDialog = new MaterialDialog.Builder(context)
                .title(R.string.text_run_repeatedly)
                .customView(binding.getRoot(), true)
                .positiveText(R.string.ok)
                .onPositive((dialog, which) -> startScriptRunningLoop())
                .build();
    }

    private void startScriptRunningLoop() {
        try {
            int loopTimes = Integer.parseInt(binding.loopTimes.getText().toString());
            float loopInterval = Float.parseFloat(binding.loopInterval.getText().toString());
            float loopDelay = Float.parseFloat(binding.loopDelay.getText().toString());
            Scripts.INSTANCE.runRepeatedly(mScriptFile,
                    loopTimes,
                    (long) (1000L * loopDelay),
                    (long) (loopInterval * 1000L));
        } catch (NumberFormatException e) {
            GlobalAppContext.toast(R.string.text_number_format_error);
        }
    }

    public ScriptLoopDialog windowType(int windowType) {
        Window window = mDialog.getWindow();
        if (window != null) {
            window.setType(windowType);
        }
        return this;
    }

    public void show() {
        DialogUtils.showDialog(mDialog);
    }
}

File: app/src/main/java/org/autojs/autojs/ui/common/ScriptLoopDialog.kt
package org.autojs.autojs.ui.common

import android.content.Context
import com.afollestad.materialdialogs.MaterialDialog
import com.stardust.app.DialogUtils
import com.stardust.app.GlobalAppContext
import org.autojs.autojs.R
import org.autojs.autojs.databinding.DialogScriptLoopBinding
import org.autojs.autojs.model.script.ScriptFile
import org.autojs.autojs.model.script.Scripts

/**
 * Created by Stardust on 2017/7/8.
 */

class ScriptLoopDialog(context: Context, private val scriptFile: ScriptFile) {

    private val binding =
        DialogScriptLoopBinding.inflate(android.view.LayoutInflater.from(context), null, false)
    private val dialog = MaterialDialog.Builder(context)
        .title(R.string.text_run_repeatedly)
        .customView(binding.root, true)
        .positiveText(R.string.ok)
        .onPositive { _, _ -> startScriptRunningLoop() }
        .build()

    private fun startScriptRunningLoop() {
        try {
            val loopTimes = binding.loopTimes.text.toString().toInt()
            val loopInterval = binding.loopInterval.text.toString().toFloat()
            val loopDelay = binding.loopDelay.text.toString().toFloat()
            Scripts.runRepeatedly(
                scriptFile,
                loopTimes,
                (1000L * loopDelay).toLong(),
                (loopInterval * 1000L).toLong()
            )
        } catch (_: NumberFormatException) {
            GlobalAppContext.toast(R.string.text_number_format_error)
        }
    }

    fun windowType(windowType: Int): ScriptLoopDialog {
        val window = dialog.window
        if (window != null) {
            window.setType(windowType)
        }
        return this
    }

    fun show() {
        DialogUtils.showDialog(dialog)
    }

}
